pallets/interest-accrual/src/lib.rs

```diff
@@ -241,47 +263,40 @@ pub mod pallet {
 			interest_rate_per_sec: T::InterestRate,
 			normalized_debt: T::Balance,
 		) -> Result<T::Balance, DispatchError> {
-			Rate::<T>::get(interest_rate_per_sec)
-				.ok_or(Error::<T>::NoSuchRate)
-				.and_then(|rate| {
-					Self::calculate_debt(normalized_debt, rate.accumulated_rate)
-						.ok_or(Error::<T>::DebtCalculationFailed)
-				})
-				.map_err(Into::into)
+			let rate = Self::get_rate(interest_rate_per_sec)?;
+			let debt = Self::calculate_debt(normalized_debt, rate.accumulated_rate)
+				.ok_or(Error::<T>::DebtCalculationFailed)?;
+			Ok(debt)
 		}
 
 		pub fn get_previous_debt(
 			interest_rate_per_sec: T::InterestRate,
 			normalized_debt: T::Balance,
 			when: Moment,
 		) -> Result<T::Balance, DispatchError> {
-			if let Some(rate) = Rate::<T>::get(interest_rate_per_sec) {
-				let now = LastUpdated::<T>::get();
-				if when > now {
-					return Err(Error::<T>::NotInPast.into());
-				}
-				let delta = now - when;
-				let rate_adjustment = checked_pow(interest_rate_per_sec, delta as usize)
-					.ok_or(ArithmeticError::Overflow)?;
-				let past_rate = rate
-					.accumulated_rate
-					.checked_div(&rate_adjustment)
-					.ok_or(ArithmeticError::Underflow)?;
-				let debt = Self::calculate_debt(normalized_debt, past_rate)
-					.ok_or(Error::<T>::DebtCalculationFailed)?;
-				Ok(debt)
-			} else {
-				Err(Error::<T>::NoSuchRate.into())
+			let rate = Self::get_rate(interest_rate_per_sec)?;
+			let now = LastUpdated::<T>::get();
+			if when > now {
+				return Err(Error::<T>::NotInPast.into());
 			}
+			let delta = now - when;
+			let rate_adjustment = checked_pow(interest_rate_per_sec, delta as usize)
```

Comment:
`Moment` maybe could be `u128` in the future.
```suggestion
			let rate_adjustment = checked_pow(interest_rate_per_sec, delta.ensure_into()?)
```